src/components/AuthStoreSetter.js

```diff
@@ -0,0 +1,28 @@
+import _ from 'lodash/fp'
+import { useEffect } from 'react'
+import { useAuth } from 'react-oidc-context'
+import { processUser, reloadAuthToken } from 'src/libs/auth'
+import { useOnMount } from 'src/libs/react-utils'
+import { authStore } from 'src/libs/state'
+
+
+const AuthStoreSetter = () => {
+  const auth = useAuth()
+  useOnMount(() => authStore.update(_.set(['authContext'], auth)))
+
+  useEffect(() => {
+    return auth.events.addAccessTokenExpiring(reloadAuthToken)
+  }, [auth])
+
+  useEffect(() => {
+    return auth.events.addUserLoaded(processUser)
```

Comment:
Why are you calling both `addUserLoaded` and `addUserUnloaded`?

Reply:
These are installing listeners on user sign-in and sign-out events. We need to handle both to update our auth store.